internal/engine/eval/rego/eval.go

```diff
@@ -134,14 +138,40 @@ func (e *Evaluator) Eval(
 		return nil, fmt.Errorf("could not prepare Rego: %w", err)
 	}
 
-	rs, err := pq.Eval(ctx, rego.EvalInput(&Input{
+	input := &Input{
 		Profile:      pol,
 		Ingested:     obj,
 		OutputFormat: e.cfg.ViolationFormat,
-	}))
+	}
+
+	if err := enrichInputWithEntityProps(input, entity); err != nil {
+		return nil, fmt.Errorf("error adding properties to input: %w", err)
+	}
+
+	rs, err := pq.Eval(ctx, rego.EvalInput(input))
 	if err != nil {
 		return nil, fmt.Errorf("error evaluating profile. Might be wrong input: %w", err)
 	}
 
 	return e.reseval.parseResult(rs, entity)
 }
+
+func enrichInputWithEntityProps(
+	input *Input,
+	entity protoreflect.ProtoMessage,
+) error {
+	switch entity := entity.(type) {
+	case *minderv1.Repository:
+		input.Properties = entity.Properties
```

Comment:
Can rego evaluate protobuf? Wouldn't it be easier to use a map[string]any?

Reply:
It gets serialized to JSON, so `input` ends up being this (produced using `mindev`)
```
{
  "ingested": null,
  "output_format": "text",
  "profile": {},
  "properties": {
    "github/base_ref": "main",
    "github/clone_url": "https://github.com/blkt-test-org/demo-repo-python.git",
    "github/default_branch": "main",
    "github/deploy_url": "https://api.github.com/repos/blkt-test-org/demo-repo-python/deployments",
    "github/hook_id": 516821355,
    "github/hook_name": "web",
    "github/hook_type": "Repository",
    "github/hook_uiid": "5567daf7-6e06-4494-8a65-993f40ab040d",
    "github/hook_url": "https://api.github.com/repos/blkt-test-org/demo-repo-python/hooks/516821355",
    "github/license": "Apache-2.0",
    "github/primary_language": "",
    "github/pull_author_id": 328978,
    "github/pull_author_login": "blkt",
    "github/pull_number": 1,
    "github/pull_ref": "blkt-patch-1",
    "github/pull_sha": "b751f485143a093cd1808cce6ca83b87c72986c6",
    "github/pull_url": "https://github.com/blkt-test-org/demo-repo-python/pull/1",
    "github/repo_id": 894536760,
    "github/repo_name": "demo-repo-python",
    "github/repo_owner": "blkt-test-org",
    "github/target_url": "https://github.com/blkt-test-org/demo-repo-python.git",
    "is_archived": false,
    "is_fork": true,
    "is_private": false,
    "name": "blkt-test-org/demo-repo-python/1",
    "upstream_id": 2201191245
  }
}
```

Reply:
that's awesome and it works for me.